### PR TITLE
refactor: Moved processor discovery to yaml config

### PR DESCRIPTION
**Pull request by laf**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

Moved junos-mss to use the new yaml processor discovery.

Largely based on the sensors discovery. Changed function names to standardise. I've also abstracted some of the data into functions to re-use. I'm not sure we can have a central function for all discovery modules, I think as we add more they will all have specific quirks.

---
This change is [Reviewable](https://reviewable.io/reviews/librenms/librenms/7369)

### CODE REVIEW

**Comment by murrant** [comment]
@laf how do you handle ones that use oids like this?
```
UCD-SNMP-MIB::ssCpuUser.0 = INTEGER: 12
UCD-SNMP-MIB::ssCpuSystem.0 = INTEGER: 6
UCD-SNMP-MIB::ssCpuIdle.0 = INTEGER: 81
```
To get usage I would have to add ssCpuUser and ssCpuSystem or do 100 - ssCpuIdle.  (this same kind of issue will come up with memory too)

Also, I don't think multiplier is applicable here.

**Comment by laf** [comment]
I don't handle those. I'm expecting the function to develop as we come across different scenarios and in some cases, that we can't do it via yaml. 

Mempools will be completely different anyway. We need free, used and max + storing those oids which we don't do yet. 

Multiplier is in as it's using a central function and defaults to 1 anyway so doesn't effect the value.

**Comment by laf** [comment]
bump

**Comment by murrant** [comment]
I'm still a little unsure here.  You are trying to reuse the generic code (a good thing) but I'm not sure they are similar enough.

But, perhaps the code can be made flexible enough for this without making it too complicated and allow for other things like memory easily...

I know really unhelpful.

**Comment by laf** [comment]
My goal right now, care less about the current code base and get things abstracted away so we can switch away from the current code base. I'm happy to make sure things are tidy but I don't want to spend more time trying to make it perfect.

**Comment by murrant** [comment]
Yeah, the yaml format is the most important part really.

@laf can you try to add some of the more complicated cpu usage scenarios?  I would feel a lot better about it then.

**Comment by laf** [comment]
@murrant I really don't have time to. I'm happy to expand the function out when the time comes.

**Comment by kkrumm1** [comment]
is this ready for a test pull? @laf @murrant

**Comment by laf** [comment]
@kkrumm1 As far as I'm concerned yes. It does only switch juniper-mss to use the yaml file but in general does make some core code changes so it would be good to check it doesn't break anything for anyone outside of my setup.

**Comment by kkrumm1** [comment]
testing it on my home set up and then I will give a try at work.

**Comment by kkrumm1** [comment]
Tested at home and didn't see any issues. I'm going to now give it a try at work.

**Comment by kkrumm1** [comment]
Tested at work and didn't see any issues.

**Comment by laf** [comment]
Bump?

**Comment by laf** [comment]
@librenms/reviewers anyone?

**Comment by laf** [comment]
@librenms/reviewers Bumping again. I have to rebase this but don't want to do so unless it's going to get merged but at this rate I'm more up for closing it.

**Comment by Rosiak** [comment]
Haven't looked at the code but just tested this at home, doesn't seem to break anything.

File: doc/Developing/os/Mem-CPU-Information.md [hunk 1]
```diff
@@ -39,9 +39,54 @@ if (is_numeric($perc)) {
 }
 ```
 
-#### Processor
+## Processor
 
-Detection for processors is done via a single script unless custom processing of data is required (as in this example).
+Detection for processors is done via yaml for basic processor detection or a single script unless custom processing of data is required (as in this example).
+
+All yaml files are located in `includes/definitions/discovery/$os.yaml`. Defining the information hear is not always
```
Comment (Zmegolaz) [inline doc/Developing/os/Mem-CPU-Information.md]: "hear" -> "here" :)

**Comment by laf** [comment]
I've rebased this. Hoping nothing has failed in doing so.

**Comment by laf** [comment]
It's release day a week tomorrow, I'd love to get this merged in sooner rather than later.

@librenms/reviewers

**Pull request closed without merging**